### PR TITLE
Unexport APIVersionPolicy and constructor

### PR DESCRIPTION
**Pull request by chlowell**
Also adding an options bag.

### CODE REVIEW

File: sdk/azcore/runtime/policy_api_version.go [hunk 1]
```diff
@@ -24,15 +24,18 @@ const (
 	APIVersionLocationHeader = 1
 )
 
-// NewAPIVersionPolicy constructs an APIVersionPolicy. name is the name of the query parameter or header and
+// apiVersionPolicyOptions contains optional parameters for apiVersionPolicy
+type apiVersionPolicyOptions struct{}
```
Comment (jhendrixMSFT) [inline sdk/azcore/runtime/policy_api_version.go:28]: Personally, I think this is less important now that the constructor isn't exported.  It's fine to leave it too, up to you.

**Review by jhendrixMSFT — APPROVED** [review]

**Review by JeffreyRichter — APPROVED** [review]

**Comment by jhendrixMSFT** [comment]
Does the changelog need to be updated?

File: sdk/azcore/runtime/policy_api_version.go (hunk 1 above)
Comment (JeffreyRichter) [reply sdk/azcore/runtime/policy_api_version.go:28]: Agreed, your call Charles.

**Comment by chlowell** [comment]
No, the only part of this in the changelog is `ClientOptions.APIVersion`

File: sdk/azcore/runtime/policy_api_version.go (hunk 1 above)
Comment (chlowell) [reply sdk/azcore/runtime/policy_api_version.go]: ✂️✂️✂️

**Pull request merged**